testing/smoke-ide-test/src/smokeIdeTest/groovy/org/gradle/ide/sync/AbstractIdeSyncTest.groovy

```diff
@@ -51,18 +53,31 @@ abstract class AbstractIdeaSyncTest extends Specification {
         return IntegrationTestBuildContext.INSTANCE
     }
 
+    /**
+     * Runs a full sync process for the build-under-test with a given Android Studio version.
+     * <p>
+     */
+    protected void androidStudioSync(String version) {
+        ideSync("ai-$version")
+    }
+
     /**
      * Runs a full sync process for the build-under-test with a given IntelliJ IDEA Community version.
      * <p>
      * The version can be optionally suffixed with a "build type", which is one of {@code release}, {@code rc}, {@code eap}.
      * For instance, {@code 2024.2-eap}. When the build type is not provided, it defaults to {@code release}.
      * <p>
-     * The sync runs as an external process.
-     * The IDE distribution is automatically downloaded if required.
      */
     protected void ideaSync(String version) {
+        ideSync("ic-$version")
+    }
+
+    /**
+     * Runs a full sync with a given IDE as an external process.
+     * The IDE distribution is automatically downloaded if required.
+     */
+    private void ideSync(String ide) {
```

Comment:
Maybe "ideWithVersion"?

Reply:
I guess Jetbrains calls it the "productCode":

 https://github.com/JetBrains/intellij-ide-starter/blob/83c513eede2c96649bdce0f8dba634cdb153fa3f/intellij.tools.ide.starter/src/com/intellij/ide/starter/models/IdeInfo.kt#L12

Reply:
`productCode` is a codified name of the desired IDE, for ex. `ic` for Idea community, `ai` for Android Studio and so on.

`ide` parameter is essentially a combination of:
- `productCode`, `version` and `releaseType` (optionally) for Idea
- `productCode`, `buildNumber` for Android Studio,

written in supported by `gradle-ide-starter` notation .

Having said that, I wouldn't say it's not totally correct to call it `ideWithVersion`

@abstratt WDYT?

Reply:
I had not heard of `gradle-ide-starter` before.

I think add a link to https://github.com/gradle/gradle-ide-starter from this class' top-level javadoc would help. Like:

> * This class provides methods to invoke `gradle-ide-starter`
> * @see https://github.com/gradle/gradle-ide-starter